### PR TITLE
IIS Modules topic updates (+in-proc updates)

### LINKED ISSUE #8682: Update IIS Modules for in-proc hosting model
In concert with updates for https://github.com/aspnet/Docs/issues/5449. 

Related: https://github.com/aspnet/Docs/issues/8613

cc: @shirhatti

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 9a93d569-16b9-a822-f9f1-ed7c3d31779f
* Version Independent ID: 4b475c3c-b698-234f-e085-a670c7e31aac
* Content: [IIS modules with ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/iis/modules?view=aspnetcore-2.1)
* Content Source: [aspnetcore/host-and-deploy/iis/modules.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/host-and-deploy/iis/modules.md)
* Product: **aspnet-core**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

### LINKED ISSUE #8804: Remove list of logging providers (leave the Logging topic link)
Remove logging provider implementations; leave the link to the Logging topic.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 9a93d569-16b9-a822-f9f1-ed7c3d31779f
* Version Independent ID: 4b475c3c-b698-234f-e085-a670c7e31aac
* Content: [IIS modules with ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/iis/modules?view=aspnetcore-2.1)
* Content Source: [aspnetcore/host-and-deploy/iis/modules.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/host-and-deploy/iis/modules.md)
* Product: **aspnet-core**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #8682 
Fixes #8804 

[Internal Review Topic](https://review.docs.microsoft.com/en-us/aspnet/core/host-and-deploy/iis/modules?view=aspnetcore-1.0&branch=pr-en-us-8831)

The jury is out on **IISClientCertificateMappingAuthentication**. It will be checked on https://github.com/aspnet/IISIntegration/issues/1432.

I also need to do some tests with the Tracing Module, which is tracked by https://github.com/aspnet/Docs/issues/8381.

Here's the list of modules from the topic with a "**No**" on them (doesn't work out-of-proc). Which ones work in-proc, if any? If unknown for a given module, I'll do some in-proc testing and see what I can come up with.

1. CGI `CgiModule`
1. HTTP Errors `CustomErrorModule`
1. Default Document `DefaultDocumentModule`
1. Directory Browsing `DirectoryListingModule`
1. File Caching `FileCacheModule`
1. HTTP Caching `HttpCacheModule`
1. Server-Side Includes `ServerSideIncludeModule`
1. Static Compression `StaticCompressionModule`
1. Static Content `StaticFileModule`

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@shirhatti 	please review

**Comment by guardrex** [comment]
@scottaddie Approved without table changes. If we find out later that modules have different behavior, we can update with the new info. Let's go ahead with these changes. **_We're on a roll here!_** :smile:

File: aspnetcore/host-and-deploy/iis/modules.md [hunk 1]
```diff
@@ -97,27 +97,27 @@ For more information on disabling modules with configuration settings, follow th
...
+    <system.webServer>
+      <modules>
+        <remove name="MODULE_NAME" />
+      </modules>
+    </system.webServer>
    </configuration>
    ```
 
-An IIS module can also be removed with *Appcmd.exe*. Provide the `MODULE_NAME` and `APPLICATION_NAME` in the command:
+An IIS module can also be removed with `Appcmd.exe`. Provide the `MODULE_NAME` and `APPLICATION_NAME` in the command:
```
Comment (scottaddie) [inline aspnetcore/host-and-deploy/iis/modules.md]: Use italics for file names.

Comment (guardrex) [reply aspnetcore/host-and-deploy/iis/modules.md]: I guess I was thinking ... 'this is a command' ... but yes ... that was incorrect.

**Pull request merged**